src/main/java/org/opentripplanner/model/plan/ScheduledTransitLeg.java

```diff
@@ -261,10 +279,29 @@ public List<StopArrival> getIntermediateStops() {
     for (int i = boardStopPosInPattern + 1; i < alightStopPosInPattern; i++) {
       StopLocation stop = tripPattern.getStop(i);
 
+      final var arrivalTime = ServiceDateUtils.toZonedDateTime(
+        serviceDate,
+        zoneId,
+        tripTimes.getArrivalTime(i)
+      );
+      final var departureTime = ServiceDateUtils.toZonedDateTime(
+        serviceDate,
+        zoneId,
+        tripTimes.getDepartureTime(i)
+      );
+
+      var arrival = LegTime.ofStatic(arrivalTime);
+      var departure = LegTime.ofStatic(departureTime);
+
+      if (getRealTime()) {
+        arrival = LegTime.of(arrivalTime, tripTimes.getArrivalDelay(i));
+        departure = LegTime.of(departureTime, tripTimes.getDepartureDelay(i));
+      }
+
```

Comment:
Consider delegating this to a LegTimeMapper, e.g.`map(tripTimes, i, getRealTime())`

Reply:
Implemented in https://github.com/opentripplanner/OpenTripPlanner/pull/5660/commits/91989e297ee37707302f09ede6b7ab4c18032c78